newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -78,15 +86,26 @@ object FingerpostWireEntry extends SQLSyntaxSupport[FingerpostWireEntry] {
       rs.long(fm.id),
       rs.string(fm.externalId),
       rs.zonedDateTime(fm.ingestedAt),
-      Json.parse(rs.string(fm.column("content"))).as[FingerpostWire]
+      Json.parse(rs.string(fm.column("content"))).as[FingerpostWire],
+      rs.stringOpt(fm.column("highlight"))
     )
 
   private def clamp(low: Int, x: Int, high: Int): Int =
     math.min(math.max(x, low), high)
 
-  def get(id: Int): Option[FingerpostWireEntry] = DB readOnly {
+  def get(
+      id: Int,
+      maybeFreeTextQuery: Option[String] = None
+  ): Option[FingerpostWireEntry] = DB readOnly {
+    val contentCol = FingerpostWireEntry.syn.column("content")
+    val highlightsClause = maybeFreeTextQuery match {
+      case Some(query) =>
+        sqls""", ts_headline('simple', $contentCol->>'body_text', websearch_to_tsquery($query), 'StartSel=<mark>, StopSel=</mark>, HighlightAll=true') AS h_on_fm"""
```

Comment:
searching on 'simple' will return only matches that match exactly - if there is something that has a matching stem then the highlighter won't find anything to return and the results look odd; try searching for 'Londoner' - you'll get lots of results for 'London', but because it doesn't match the highlight is just for the first few words in the body

This might be the desired behaviour, in which case we should not display the highlight; but I think it'd be best if we used the 'english' "dictionary" -- even if the results aren't exactly what the user entered, they can at least see what we thought was similar enough to return this wire (and can complain if it's nonsensical and we can try to improve)

```suggestion
        sqls""", ts_headline('english', $contentCol->>'body_text', websearch_to_tsquery('english', $query), 'StartSel=<mark>, StopSel=</mark>, HighlightAll=true') AS h_on_fm"""
```

Reply:
ah, fantastic, thanks! I had a todo to try and work out why it was sometimes returning just the first few lines, but hadn't got round to it yet!